src/main/java/friday/Deadline.java

```diff
@@ -0,0 +1,51 @@
+package friday;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents a deadline task in Friday.
+ */
+public class Deadline extends Task {
+    protected String deadlineStr;
+    protected LocalDate deadline;
+
+    /**
+     * Constructs a new Deadline task.
+     *
+     * @param description The description of the event.
+     * @param deadlineStr The deadline of the event.
+     */
+    public Deadline(String description, String deadlineStr) {
+        super(description);
+        this.deadlineStr = deadlineStr;
+    }
+
+    /**
+     * Formats the deadline string
```

Comment:
Just a small nit regarding Javadocs:

```suggestion
     * Formats the deadline string.
```